lib/spack/spack/directives.py

```diff
@@ -422,7 +422,7 @@ def _execute_version(pkg):
     return _execute_version
 
 
-def _depends_on(pkg, spec, when=None, type=default_deptype, patches=None):
+def pkg_depends_on(pkg, spec, when=None, type=default_deptype, patches=None):
```

Comment:
Is this stuff related to one of your other PRs?

Reply:
Yes, that's #37380